app/plugins/adp/src/hooks/useCatalogUsersList.tsx

```diff
@@ -0,0 +1,76 @@
+import { useState, useEffect } from 'react';
+import { useApi, errorApiRef } from '@backstage/core-plugin-api';
+import { catalogApiRef } from '@backstage/plugin-catalog-react';
+import type { Entity, UserEntityV1alpha1 } from '@backstage/catalog-model';
+
+type CatalogUsersListOptions = {
+  label: string;
+  value: string;
+};
+
+export const useCatalogUsersList = (): { label: string; value: string }[] => {
+  const [options, setOptions] = useState<CatalogUsersListOptions[]>([]);
+
+  const errorApi = useApi(errorApiRef);
+  const catalog = useApi(catalogApiRef);
+
+  useEffect(() => {
+    (async () => {
+      try {
+        const userEntities = await catalog.getEntities({
```

Comment:
This will cause the `getEntities` endpoint to be re-called every time the catalog or errorApi change. Im not sure if thats what we intend to happen. I added the `useAsyncDataSource` hook to prevent the api from being called more often than needed, do you think we need this here?